rfcs/0001-meta-schedule-autotensorir.md

```diff
@@ -0,0 +1,604 @@
+
+This subsection shows that the design space induced by TE manual schedule, AutoTVM and AutoScheduler
+are all subsets of meta schedule, and meta schedule further allows mixing those three styles to
+search jointly.
+
+- **Manual schedule**. The TE schedule is a special case of a meta schedule program, where there is no
+randomness introduced by sampling instructions. It is a single point in terms of design space.
+- **AutoTVM (Template-based tuning)**. It is more natural representation of AutoTVM’s schedule
+templates (knobs) by writing one or more schedule functions in meta schedule with sampling
+instructions. The probability space supported by the sampling instructions is the design space to
+be explored.
+- **AutoScheduler (Template-free tuning)**. As mentioned in the previous section, application
+  of composite schedule rules generates the design space, which is equivalent to AutoScheduler’s
+  sketch generation.
+- **Mixing styles in design space definition**. By taking union of the spaces induced by the three
+  special cases, meta schedule allows developers to combine generic rules that AutoScheduler
+  provides and operator-specific scheduling.
+
+## 4. Reference-level explanation
+
+This section introduces the underlying techniques for the automation system to extract and
+explore the design space. The figure below briefly illustrates the workflow of the system:
+
+![meta-schedule-workflow](../resources/meta-schedule-workflow.png)
+
+### 4.1. Execution trace as the design space
+
+**Trace**. To represent the design space defined by the meta schedule DSL, the underlying system
+records all the instructions users applied to the schedule class, including sampling and schedule
+primitives. This list of scheduling instructions being invoked, along with the random decisions made
+on sampling instructions, is called a trace.
+
+For instance, executing the example above results in the following trace:
+
+```
+Instruction 0. Sample-Perfect-Tile
+Instruction 1. Sample-Perfect-Tile
+Instruction 2. Sample-Perfect-Tile
+Instruction 3. Split
+Instruction 4. Split
+Instruction 5. Split
+Instruction 6. Reorder
+```
+
+**Trace forms design space.** A trace may contain zero or more sampling instructions, which
+introduces the uncertainty in scheduling - one instance of sampling results in one point in the
+design space. Therefore, the trace itself forms a design space to explore, e.g. which set of tile
+sizes works best on a specific hardware.
+
+**Union of design space**. Meta schedule works on a set of traces, representing the union of the design
+spaces represented by every single trace.
+
+**Fork a trace**. When two different decisions in the scheduling process are equally important to
+generate high-performance schedules, it is allowed to fork the trace into two, and the design space is
+the union of the forked traces.
+
+The trace is not strictly user-facing, but can be accessed and printed with the following syntax:
```

Comment:
if this isn't user-facing, how is the user supposed to invoke Meta Scheduler in a repeatable way? is there a serialization mechanism provided for the implementation (in particular `decisions` dict below)?